server/src/main/java/org/opensearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -332,11 +334,18 @@ public PublicationContext newPublicationContext(
         boolean isRemotePublicationEnabled,
         PersistedStateRegistry persistedStateRegistry
     ) {
-        final PublicationContext publicationContext = new PublicationContext(
-            clusterChangedEvent,
-            isRemotePublicationEnabled,
-            persistedStateRegistry
-        );
+        if (isRemotePublicationEnabled == true) {
+            if (allNodesRemotePublicationEnabled.get() == false) {
+                if (validateRemotePublicationOnAllNodes(clusterChangedEvent.state().nodes()) == true) {
+                    allNodesRemotePublicationEnabled.set(true);
+                }
+            }
+            if (allNodesRemotePublicationEnabled.get() == true) {
+                // if all nodes are remote then create remote publication context
+                return new RemotePublicationContext(clusterChangedEvent, persistedStateRegistry);
+            }
+        }
+        final PublicationContext publicationContext = new PublicationContext(clusterChangedEvent, persistedStateRegistry);
```

Comment:
Sorry why is this needed. Can we publish cluster state dual mode based on the destination node attribute?